writer/json_test.go

```diff
@@ -0,0 +1,122 @@
+package writer
+
+import (
+	"bytes"
+	"encoding/json"
+	"testing"
+
+	"github.com/dineshba/tf-summarize/terraformstate"
+	. "github.com/hashicorp/terraform-json"
+	"github.com/nsf/jsondiff"
+)
+
+// Mock data for testing multiple resources
+func createMockResourceChanges() []terraformstate.ResourceChanges {
+	return []terraformstate.ResourceChanges{
+		{
+			{
+				Address: "module.test.azapi_resource.logical_network",
+				Type:    "aws_instance",
+				Name:    "example",
+				Change: &Change{
+					Before:  map[string]interface{}{"name": "old_instance"},
+					After:   map[string]interface{}{"name": "new_instance"},
+					Actions: Actions{ActionCreate},
+				},
+			},
+		},
+		{
+			{
+				Address: "module.test.aws_s3_bucket.example",
+				Type:    "aws_s3_bucket",
+				Name:    "example",
+				Change: &Change{
+					Before:  map[string]interface{}{"name": "old_bucket"},
+					After:   map[string]interface{}{"name": "new_bucket"},
+					Actions: Actions{ActionUpdate},
+				},
+			},
+		},
+		{
+			{
+				Address: "module.test.aws_security_group.example",
+				Type:    "aws_security_group",
+				Name:    "example",
+				Change: &Change{
+					Before:  map[string]interface{}{"name": "old_sg"},
+					After:   map[string]interface{}{"name": "new_sg"},
+					Actions: Actions{ActionDelete},
+				},
+			},
+		},
+	}
+}
+
+func TestJSONWriter(t *testing.T) {
+	mockResourceChangesArray := createMockResourceChanges()
+
+	expectedOutputs := []map[string]interface{}{
+		{
+			"module": map[string]interface{}{
+				"test": map[string]interface{}{
+					"azapi_resource": map[string]interface{}{
+						"logical_network": map[string]interface{}{
+							"(+)": map[string]interface{}{
+								"name": "new_instance",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			"module": map[string]interface{}{
+				"test": map[string]interface{}{
+					"aws_s3_bucket": map[string]interface{}{
+						"example": map[string]interface{}{
+							"(~)": map[string]interface{}{
+								"name": map[string]interface{}{
+									"changed": []string{
+										"old_bucket",
+										"new_bucket",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			"module": map[string]interface{}{
+				"test": map[string]interface{}{
+					"aws_security_group": map[string]interface{}{
+						"example": map[string]interface{}{
+							"(-)": map[string]interface{}{
+								"name": "old_sg",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for i, changes := range mockResourceChangesArray {
```

Comment:
It is better to always group all inputs into one struct for tests. Is https://www.digitalocean.com/community/tutorials/how-to-write-unit-tests-in-go-using-go-test-and-the-testing-package#step-4-mdash-writing-table-driven-tests-in-go useful here?

Reply:
Done